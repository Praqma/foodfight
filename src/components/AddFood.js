import React from 'react';
import { connect } from "react-redux";
import { addFood } from "../redux/actions";

export class AddFood extends React.Component {
    constructor(props){
        super(props);
        this.state = {name:""}
      }
    
      updateName = name => {
        this.setState({name});
      };
    
      handleAddFood = (event) => {
        this.props.addFood({name:this.state.name});
        event.preventDefault();
        this.setState({ name: "" });
      };
    
    render(){
        return(
        <form onSubmit={this.handleAddFood}>
        <label>Food to add:
          <input type="text" value={this.state.name} onChange={e => this.updateName(e.target.value)}/>
        </label>
        <input type="submit" value="Add food" />
        </form>
        )
    }

}
export default connect(
    null,
    { addFood }
  )(AddFood)