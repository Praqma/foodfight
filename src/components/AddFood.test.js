import React from 'react';
import {AddFood} from './AddFood';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

test('has a form', () => {
    let store = mockStore({});    
    const component  = renderer.create(
        <Provider store={store}>
            <AddFood/>
        </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})  
