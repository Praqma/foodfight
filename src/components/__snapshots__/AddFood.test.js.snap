// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFood has a form 1`] = `
<div>
  <label>
    Food to add:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    className="add-food"
    onClick={[Function]}
  >
    Add Food
  </button>
</div>
`;
